### PR TITLE
Fixes a TypeError in GRPC::Core::Channel#connectivity_state

### PR DESCRIPTION
**Pull request by yugui**
The method used to always raise a TypeError:

>  no implicit conversion of false into Integer

This change also corrects the rdoc of the method.

### CODE REVIEW

**Comment by apolcyn** [comment]
jenkins: this is ok to test

**Comment by apolcyn** [comment]
thanks for the fix! This is probably outside the scope of this PR but I think there's another issue with the `grpc_rb_channel_get_connectivity_state` function. This can be done in a separate follow up if you prefer but I'd like to further change it with something like:

```
 static VALUE grpc_rb_channel_get_connectivity_state(int argc, VALUE *argv,
                                                     VALUE self) {
-  VALUE try_to_connect = Qfalse;
+  VALUE try_to_connect_param = Qfalse;
+  int grpc_try_to_connect = 0;
   grpc_rb_channel *wrapper = NULL;
   grpc_channel *ch = NULL;
 
   /* "01" == 0 mandatory args, 1 (try_to_connect) is optional */
-  rb_scan_args(argc, argv, "01", try_to_connect);
+  rb_scan_args(argc, argv, "01", &try_to_connect_param);
+  grpc_try_to_connect = try_to_connect_param == Qtrue? 1 : 0;
 
   TypedData_Get_Struct(self, grpc_rb_channel, &grpc_channel_data_type, wrapper);
   ch = wrapper->wrapped;
@@ -197,8 +199,9 @@ static VALUE grpc_rb_channel_get_connectivity_state(int argc, VALUE *argv,
     rb_raise(rb_eRuntimeError, "closed!");
     return Qnil;
   }
   return LONG2NUM(
-      grpc_channel_check_connectivity_state(ch, (int)try_to_connect));
+      grpc_channel_check_connectivity_state(ch, grpc_try_to_connect));
 }
```

**Comment by apolcyn** [comment]
ping on this - I think it would be good to get this in, unless there is more work in progress we can merge and post a follow up PR

**Comment by yugui** [comment]
@apolcyn Sorry for my late reply.

Personally I prefer putting your suggestion into this PR because both of the fixes make  `connectivity_state` method work as described.

But I've applied your suggestion in another way -- `RTEST` macro is the canonical way to convert boolean values in Ruby into C.

File: src/ruby/ext/grpc/rb_channel.c [hunk 1]
```diff
@@ -175,11 +175,13 @@ static VALUE grpc_rb_channel_init(int argc, VALUE *argv, VALUE self) {
...
-    creds = ...
-    secure_channel = Channel:new("myhost:443", {'arg1': 'value1'}, creds)
+    ch.connectivity_state       -> state
+    ch.connectivity_state(true) -> state
 
-  Creates channel instances. */
+  Indicates the current state of the channel, whose value is one of the
+  constants defined in GRPC::Core::ConnectivityStates.
+
+  It also tries to connect if the chennel is idle in the second form. */
```
Comment (apolcyn) [inline src/ruby/ext/grpc/rb_channel.c]: thanks, one more nit: I think line 192 needs to be changed to:

`rb_scan_args(argc, argv, "01", &try_to_connect);`

the change being +`&try_to_connect` and  -`try_to_connect`

**Comment by apolcyn** [comment]
@yugui thanks for the issues that were pointed out in this PR. changes and fixes here were included in https://github.com/grpc/grpc/pull/9986 and are now released in the latest 1.2.2 release

**Pull request closed without merging**